src/platform/forms-system/src/js/web-component-fields/VaCheckboxGroupField.jsx

```diff
@@ -94,19 +97,30 @@ export default function VaCheckboxGroupField(props) {
         {checkboxes?.length > 0 &&
           // eslint-disable-next-line no-unused-vars
           checkboxes.map(({ key, uiSchema, schema, formData }) => {
-            return (
+            const elmProps = {
+              key,
+              'data-key': key,
+              name: `${props.childrenProps.idSchema.$id}_${key}`,
+              uswds: props.uiOptions?.uswds ?? true,
+              label: uiSchema?.['ui:title'] || schema.title,
+            };
+
+            return schema.type === 'boolean' ? (
               <va-checkbox
-                data-key={key}
-                name={`${props.childrenProps.idSchema.$id}_${key}`}
-                key={key}
-                uswds={props.uiOptions?.uswds ?? true}
-                label={uiSchema?.['ui:title'] || schema.title}
+                {...elmProps}
                 checked={formData === 'undefined' ? false : formData}
                 tile={props.uiOptions?.tile}
                 checkbox-description={
                   uiSchema?.['ui:description'] || schema.description
                 }
               />
+            ) : (
+              <va-text-input
```

Comment:
Added `va-text-input` support so that all the checkboxes and the input are inside a single `fieldset`

Reply:
Switched the order so that the ternary defaults to `va-checkbox`. Also, updated the `onGroupChange` function ternary to default to checked property  - this fixes some unit test failures